aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -457,174 +549,328 @@ void ImagingResampleHorizontalConvolution8u4x(
+  // lineIn + stride * (x + xmin) + 32 <= lineIn + stride * (xmax + xmin)
+  // --> x <= xmax - 32.0 / stride
+  // Strict boundary:
+  // --> x < xmax + 1 - int(ceil(32.0 / stride)) = xmax - b8_delta
+  // Soft boundary for reading inside the buffer except its boundaries:
+  // --> x < xmax + 1 - int(32.0 / stride) = xmax - b8_delta_soft
+  // RGBA: b8_delta = b8_delta_soft = 7
+  // RGB : b8_delta = 10
+  // RGB : b8_delta_soft = 9
+  const auto b8_delta = (stride == 4) ? 7 : ((is_last_line) ? 10 : 9);
+
+  // lineIn + stride * (x + xmin) + 16 <= lineIn0 + stride * (xmax + xmin)
+  // --> x <= xmax - 16.0 / stride
+  // Strict boundary:
+  // --> x < xmax + 1 - int(ceil(16.0 / stride)) = xmax - b4_delta
+  // Soft boundary for reading inside the buffer except its boundaries:
+  // --> x < xmax + 1 - int(16.0 / stride) = xmax - b4_delta_soft
+  // RGBA: b4_delta = b4_delta_soft = 3
+  // RGB : b4_delta = 5
+  // RGB : b4_delta_soft = 4
+  const auto b4_delta = (stride == 4) ? 3 : ((is_last_line) ? 5 : 4);
+
+  // lineIn0 + stride * (x + xmin) + 8 <= lineIn0 + stride * (xmax + xmin)
+  // --> x <= xmax - 8.0 / stride
+  // Strict boundary:
+  // --> x < xmax + 1 - int(ceil(8.0 / stride)) = xmax - b2_delta
+  // Soft boundary for reading inside the buffer except its boundaries:
+  // --> x < xmax + 1 - int(8.0 / stride) = xmax - b2_delta_soft
+  // RGBA: b2_delta = b2_delta_soft = 1
+  // RGB : b2_delta = 2
+  // RGB : b2_delta_soft = 1
+  const auto b2_delta = (stride == 4) ? 1 : ((is_last_line) ? 2 : 1);
+
+  const auto max_out_x_strided = out_xsize * stride;
+  const auto max_in_x_strided = in_xsize * stride;
+
+  for (const auto out_x : c10::irange(out_xsize)) {
     __m128i sss;
-    xmin = xbounds[xx * 2 + 0];
-    xmax = xbounds[xx * 2 + 1];
-    k = &kk[xx * kmax];
-    x = 0;
+    const auto xmin = idx_ptr_xmin[out_x];
+    const auto xmax = idx_ptr_size[out_x];
+    const auto * k = &kk[out_x * kmax];
+    int64_t x = 0;
 
     if (xmax < 8) {
       sss = _mm_set1_epi32(1 << (coefs_precision - 1));
     } else {
       // Lower part will be added to higher, use only half of the error
-      __m256i sss256 = _mm256_set1_epi32(1 << (coefs_precision - 2));
-
-      for (; x < xmax - 7; x += 8) {
-        __m256i pix, mmk, source;
-        __m128i tmp = _mm_loadu_si128((__m128i*)&k[x]);
-        __m256i ksource =
-            _mm256_insertf128_si256(_mm256_castsi128_si256(tmp), tmp, 1);
-
-        // clang-format off
-        source = _mm256_loadu_si256((__m256i*)&lineIn[x + xmin]);
-        pix = _mm256_shuffle_epi8(source, _mm256_set_epi8(
-          -1,7, -1,3, -1,6, -1,2, -1,5, -1,1, -1,4, -1,0,
-          -1,7, -1,3, -1,6, -1,2, -1,5, -1,1, -1,4, -1,0));
-        mmk = _mm256_shuffle_epi8(ksource, _mm256_set_epi8(
-          11,10, 9,8, 11,10, 9,8, 11,10, 9,8, 11,10, 9,8,
-          3,2, 1,0, 3,2, 1,0, 3,2, 1,0, 3,2, 1,0));
+      auto sss256 = _mm256_set1_epi32(1 << (coefs_precision - 2));
```

Comment:
Can you make this a stack PR with style changes as a separate PR. It would make the diff much nicer to read.

Reply:
Closed in favor of this stack: https://github.com/pytorch/pytorch/pull/96848